Update Host section in Fundamentals Overview (#12816)
aspnet/AspNetCore.Docs: aspnetcore/fundamentals/index.md

The `Startup` class is where:

* Services required by the app are configured.
* The request handling pipeline is defined.

> *Services* are components that are used by the app. For example, an Entity Framework Core context object is a service. Code to configure (or *register*) services is added to the `Startup.ConfigureServices` method.

Reviewer: ```suggestion
*Services* are components that are used by the app. Code to configure (or *register*) services is added to the `Startup.ConfigureServices` method.
```

Alternatively, is there something more general/generic that could be used for the service example? Even something like MVC, which is widely understood as an architectural pattern provided by an app framework.

Author: > is there something more general/generic that could be used for the service example?

MVC could be misleading, as it sounds like a pattern than a component.  I'm open to other ideas ... I could refer to a logger, that would be more general than EF.

Reviewer: Just a passing thought. EF is probably a cryptic tech for some readers. I like the logger idea.

The main reason for including all of the app's interdependent resources in one object is lifetime management: control over app startup and graceful shutdown.

::: moniker range=">= aspnetcore-3.0"

> Two host versions are available: Generic Host and Web Host. In ASP.NET Core 3.0 and later, Generic Host is recommended, and Web Host is available only for backwards compatibility.

Reviewer: Since this is versioned content, it's probably ok to just state it ...

```suggestion
Two hosts are available: Generic Host and Web Host. The Generic Host is recommended, and the Web Host is available for backwards compatibility.
```

> The code to create Generic Host is in `Program.Main`:

Reviewer: ```suggestion
The code to create a Generic Host is in `Program.Main`:
```

[!code-csharp[](index/snapshots/3.x/Program1.cs)]

The `CreateDefaultBuilder` and `ConfigureWebHostDefaults` methods configure a host with commonly used options, such as the following:

* Use [Kestrel](#servers) as the web server and enable IIS integration.
* Load configuration from *appsettings.json*, *appsettings.{Environment Name}.json*, environment variables, command line arguments, and other configuration sources.
* Send logging output to the console and debug providers.

For more information, see <xref:fundamentals/host/generic-host>.

::: moniker-end

::: moniker range="< aspnetcore-3.0"

> Two host versions are available: Web Host and Generic Host. In ASP.NET Core 2.x, Generic Host is only for non-web scenarios.

Reviewer: ```suggestion
Two hosts are available: Web Host and Generic Host. The Web Host is recommended, and the Generic Host is only for non-web scenarios.
```

> The code to create Web Host is in `Program.Main`:

Reviewer: ```suggestion
The code to create a Web Host is in `Program.Main`:
```

[!code-csharp[](index/snapshots/2.x/Program1.cs)]

The `CreateDefaultBuilder` method configures a host with commonly used options, such as the following:

* Use [Kestrel](#servers) as the web server and enable IIS integration.
* Load configuration from *appsettings.json*, *appsettings.{Environment Name}.json*, environment variables, command line arguments, and other configuration sources.
* Send logging output to the console and debug providers.

For more information, see <xref:fundamentals/host/web-host>.

::: moniker-end

### Non-web scenarios

> Generic Host allows other types of apps to use cross-cutting framework extensions, such as logging, DI, configuration, and app lifetime management. For more information, see <xref:fundamentals/host/generic-host> and <xref:fundamentals/host/hosted-services>.

Reviewer: ```suggestion
The Generic Host allows other types of apps to use cross-cutting framework extensions, such as logging, dependency injection (DI), configuration, and app lifetime management. For more information, see <xref:fundamentals/host/generic-host> and <xref:fundamentals/host/hosted-services>.
```
